silx/gui/plot/items/image.py

```diff
@@ -459,3 +458,73 @@ class MaskImageData(ImageData):
     internal silx widgets.
     """
     pass
+
+
+class ImageStack(ImageData):
+    """Item to store a stack of images and to show it in the plot as one
+    of the images of the stack.
+
+    The stack is a 3D array ordered this way: `frame id, x, y`.
+    So the first image of the stack can be reached this way: `stack[0, :, :]`
+    """
+
+    def __init__(self):
+        ImageData.__init__(self)
+        self.__stack = None
+        """A 3D numpy array (or a mimic one, see ListOfImages)"""
+        self.__stackPosition = None
+        """Displayed position in the cube"""
+
+    def setStack(self, stack, position=None):
```

Comment:
I'd call it `setStackData`, in the items most if not all method taking numpy.array of data have names ends with `Data`.
Also, I'd put a `copy=True` argument with a pass-through if False and a copy if True

Reply:
Ok. I am not sure for the copy=True, cause we talk about a lot of data, which can still be in files. But it's fine, it's much better to be consistent.

Reply:
Yes, you always want to put `copy=False`, but I prefer it to be explicit.